lib/broccoli/broccoli-typescript.js

```diff
@@ -1,327 +1,170 @@
-var originalEmitFiles = ts.emitFiles;
-ts.emitFiles = function (resolver, host, targetSourceFile) {
-    if (tsEmitInternal) {
-        var orignalgetCompilerOptions_1 = host.getCompilerOptions;
-        host.getCompilerOptions = function () {
-            var options = clone(orignalgetCompilerOptions_1.call(host));
-            options.stripInternal = false;
-            options.outDir = options.outDir + "/" + exports.INTERNAL_TYPINGS_PATH;
-            return options;
-        };
-    }
-    return originalEmitFiles(resolver, host, targetSourceFile);
-};
-/**
- * Broccoli plugin that implements incremental Typescript compiler.
- *
- * It instantiates a typescript compiler instance that keeps all the state about the project and
- * can re-emit only the files that actually changed.
- *
- * Limitations: only files that map directly to the changed source file via naming conventions are
- * re-emitted. This primarily affects code that uses `const enum`s, because changing the enum value
- * requires global emit, which can affect many files.
- */
-var DiffingTSCompiler = (function () {
-    function DiffingTSCompiler(inputPath, cachePath, options) {
-        this.inputPath = inputPath;
-        this.cachePath = cachePath;
-        this.options = options;
-        this.fileRegistry = Object.create(null);
-        this.firstRun = true;
-        this.previousRunFailed = false;
-        // Whether to generate the @internal typing files (they are only generated when `stripInternal` is
-        // true)
-        this.genInternalTypings = false;
-        if (options.rootFilePaths) {
-            this.rootFilePaths = options.rootFilePaths.splice(0);
-            delete options.rootFilePaths;
-        }
-        else {
-            this.rootFilePaths = [];
-        }
-        if (options.internalTypings) {
-            this.genInternalTypings = true;
-            delete options.internalTypings;
-        }
-        // the conversion is a bit awkward, see https://github.com/Microsoft/TypeScript/issues/5276
-        // in 1.8 use convertCompilerOptionsFromJson
-        this.tsOpts =
-            ts.parseJsonConfigFileContent({ compilerOptions: options, files: [] }, null, null).options;
-        if (this.tsOpts.stripInternal === false) {
-            // @internal are included in the generated .d.ts, do not generate them separately
-            this.genInternalTypings = false;
-        }
-        // TODO: the above turns rootDir set to './' into an empty string - looks like a tsc bug
-        //       check back when we upgrade to 1.7.x
-        if (this.tsOpts.rootDir === '') {
-            this.tsOpts.rootDir = './';
-        }
-        this.tsOpts.outDir = this.cachePath;
-        this.tsServiceHost = new CustomLanguageServiceHost(this.tsOpts, this.rootFilePaths, this.fileRegistry, this.inputPath);
-        this.tsService = ts.createLanguageService(this.tsServiceHost, ts.createDocumentRegistry());
+/* jshint node: true, esnext: true */
```

Comment:
Yeah I just saw it hasn't been updated in 2 years. Actually maybe https://www.npmjs.com/package/broccoli-typescript-compiler is better. Could you take a look?